src/AltinnCore/Common/Services/Implementation/DataAppSI.cs

```diff
@@ -75,55 +97,72 @@ public async Task<Instance> InsertData<T>(T dataToSerialize, Guid instanceId, Ty
         /// <inheritdoc />
         public void UpdateData<T>(T dataToSerialize, Guid instanceId, Type type, string applicationOwnerId, string applicationId, int instanceOwnerId, Guid dataId)
         {
-            string apiUrl = $"{_platformSettings.GetApiStorageEndpoint}instances/{instanceId}/data/{dataId}?instanceOwnerId={instanceOwnerId}";
+            string apiUrl = $"instances/{instanceId}/data/{dataId}?instanceOwnerId={instanceOwnerId}";
+            string token = JwtTokenUtil.GetTokenFromContext(_httpContextAccessor.HttpContext, _cookieOptions.Cookie.Name);
+            _client.DefaultRequestHeaders.Accept.Clear();
 
-            using (HttpClient client = new HttpClient())
+            if (_client.DefaultRequestHeaders.Contains("Authorization"))
             {
-                client.BaseAddress = new Uri(apiUrl);
-                XmlSerializer serializer = new XmlSerializer(type);
-                using (MemoryStream stream = new MemoryStream())
+                _client.DefaultRequestHeaders.Remove("Authorization");
+            }
+
+            _client.DefaultRequestHeaders.Add("Authorization", "Bearer " + token);
+
+            XmlSerializer serializer = new XmlSerializer(type);
+            using (MemoryStream stream = new MemoryStream())
+            {
+                serializer.Serialize(stream, dataToSerialize);
+                stream.Position = 0;
+                StreamContent streamContent = new StreamContent(stream);
+                streamContent.Headers.ContentType = MediaTypeHeaderValue.Parse("application/xml");
+
+                if (_client.DefaultRequestHeaders.Contains("Authorization"))
```

Comment:
this check is already done above

Reply:
It's supposed to be for the streamcontent and not the client, updated now. 
However, do we actually need to add this token to the streamcontent as well? I haven't been able to figure that out.